File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       HashSet<Integer> set=new HashSet<>();
       int result[]=new int[2];
       int num=-1;
       for(int i=0;i<nums.length;i++)
       {
           if(set.contains(target-nums[i])){
               result[1]=i;
               num=target-nums[i];
               break;
           }else{
               set.add(nums[i]);
           }
       }
       for(int i=0;i<nums.length;i++)
       if(nums[i]==num){
       result[0]=i;
       break;
       }
       return result;
    }
}

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    resmap:=map[int]bool{}

    index1:=-1
    index2:=-1

    dataIndexToFind:=-1

    for i,num:=range(nums){
        diff:=target-num // 9-7=2

        _,exist:=resmap[diff]
        if exist{
            index1=i // 1
            dataIndexToFind=diff //2
            break;
        }else{
            resmap[num]=true
        }
    }

    for i,data:=range(nums){
        if data==dataIndexToFind{
            index2=i
            break
        }
    }

    return []int{index1,index2}

}